systemtest/src/test/java/io/strimzi/systemtest/upgrade/AbstractUpgradeST.java

```diff
@@ -492,13 +491,13 @@ protected void setupEnvAndUpgradeClusterOperator(ExtensionContext extensionConte
         logPodImages(clusterName);
     }
 
-    protected void verifyProcedure(JsonObject testParameters, String producerName, String consumerName, String namespace) {
-        int continuousClientsMessageCount = testParameters.getJsonObject("client").getInteger("continuousClientsMessages");
+    protected void verifyProcedure(VersionModificationData testParameters, String producerName, String consumerName, String namespace) {
+        int continuousClientsMessageCount = (int) testParameters.getClient().get("continuousClientsMessages");
```

Comment:
Can you have extra structure/class just for the client and procedures? It would simplify the getters

Reply:
i'll add getters rather than classes.